### PR TITLE
Basic path normalization added to LocateServer.h

### PR DESCRIPTION
**Pull request by JeroMiya**

### CODE REVIEW

File: inc/osvr/Client/LocateServer.h [hunk 1]
```diff
@@ -42,6 +42,15 @@ namespace osvr {
         /** @brief INTERNAL ONLY - get the current server directory, if available. */
         inline boost::optional<std::string> getServerBinaryDirectoryPath() {
             auto server = osvr::common::getEnvironmentVariable("OSVR_SERVER_ROOT");
+            if (server) {
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: Can we say `if (!server) return server;` and skip a level of indentation?

File: inc/osvr/Client/LocateServer.h [hunk 2]
```diff
@@ -42,6 +42,15 @@ namespace osvr {
...
+                std::string pathSep;
+                if (server->size() > 0) {
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: Prefer `if (!server->empty()) {`.

File: inc/osvr/Client/LocateServer.h [hunk 3]
```diff
@@ -42,6 +42,15 @@ namespace osvr {
         /** @brief INTERNAL ONLY - get the current server directory, if available. */
         inline boost::optional<std::string> getServerBinaryDirectoryPath() {
             auto server = osvr::common::getEnvironmentVariable("OSVR_SERVER_ROOT");
+            if (server) {
+                std::string pathSep;
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: Unused variable?

File: inc/osvr/Client/LocateServer.h [hunk 4]
```diff
@@ -42,6 +42,15 @@ namespace osvr {
...
+                if (server->size() > 0) {
+                    char lastChar = server->at(server->size() - 1);
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: You can use `server->back()` in C++11.

File: inc/osvr/Client/LocateServer.h [hunk 5]
```diff
@@ -42,6 +42,15 @@ namespace osvr {
...
+                    if (lastChar == '/' || lastChar == '\\') {
+                        return server->substr(0, server->size() - 1);
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: You can use `server->pop_back();` in C++11 to remove the last char from the string.

File: inc/osvr/Client/LocateServer.h [hunk 6]
```diff
@@ -50,10 +59,13 @@ namespace osvr {
             auto binPath = getServerBinaryDirectoryPath();
             if (binPath) {
 #if defined(OSVR_WINDOWS)
-                return *binPath + "\\osvr_server.exe";
+                std::string pathExtension = ".exe";
+                std::string pathSep = "\\";
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: These can be made `static` and `constexpr`.

File: inc/osvr/Client/LocateServer.h [hunk 7]
```diff
@@ -50,10 +59,13 @@ namespace osvr {
...
 #else
-                return *binPath + "/osvr_server";
+                std::string pathExtension = "";
+                std::string pathSEp = '/';
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: Use double quote for string for consistency's sake?

File: inc/osvr/Client/LocateServer.h (hunk 1 above)
Comment (JeroMiya) [inline inc/osvr/Client/LocateServer.h]: Saved two levels by also checking `server->empty()` in the same branch.

File: inc/osvr/Client/LocateServer.h (hunk 6 above)
Comment (JeroMiya) [inline inc/osvr/Client/LocateServer.h]: no `constexpr` in VS2013, so I made them `static const`.

File: inc/osvr/Client/LocateServer.h [hunk 8]
```diff
@@ -42,6 +42,13 @@ namespace osvr {
         /** @brief INTERNAL ONLY - get the current server directory, if available. */
         inline boost::optional<std::string> getServerBinaryDirectoryPath() {
             auto server = osvr::common::getEnvironmentVariable("OSVR_SERVER_ROOT");
+            if (!server || server->empty()) {
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: Nice!

File: inc/osvr/Client/LocateServer.h [hunk 9]
```diff
@@ -42,6 +42,13 @@ namespace osvr {
...
+                return server;
+            };
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: Spurious semicolon.

File: inc/osvr/Client/LocateServer.h [hunk 10]
```diff
@@ -42,6 +42,13 @@ namespace osvr {
...
+            char lastChar = server->back();
```
Comment (godbyk) [inline inc/osvr/Client/LocateServer.h]: This could be `const` if you're feeling pedantic.

**Comment by godbyk** [comment]
Looks good! 👍

**Pull request merged**